### PR TITLE
network: add tests for most of the commands

### PR DESCRIPTION
**Pull request by fyrchik**
Fixed 3 bugs:
1.  `GetBlocks` should sent exactly `Count` blocks.
2. `requestTx` couldn't didn't send anything if amount of hashes was small.
3. `requestTx` could panic, but luckily (or unluckily) (2) made this event rather unprobable.

`network.testChain` is rather ugly now, this should be fixed in #1472 . Now there are more specific requirements on what we could do with testchain in tests.

### CODE REVIEW

**Comment by roman-khimov** [comment]
965302cfc721aa30c47a4ffbac82432a89412817 is an interesting one, it might explain excessive CVs during four nodes benchmarking.

File: pkg/network/server_test.go [hunk 1]
```diff
@@ -1,17 +1,176 @@
...
+
+		require.True(t, s.consensus.(*fakeConsensus).stopped.Load())
+	})
+}
+
+func TestServerRegisterPeer(t *testing.T) {
+	const peerCount = 3
+
+	s := newTestServer(t, ServerConfig{MaxPeers: 2})
+	fakeDrop := peerDrop{newLocalPeer(t, s), nil} // fake drop is used to ensure previous channel send has finished
```
Comment (roman-khimov) [inline pkg/network/server_test.go:126]: You can `require.Eventually` instead of that.

**Review by roman-khimov — CHANGES_REQUESTED** [review]

File: pkg/network/server.go [hunk 2]
```diff
@@ -186,6 +186,7 @@ func (s *Server) Start(errChan chan error) {
 	go s.transport.Accept()
 	setServerAndNodeVersions(s.UserAgent, strconv.FormatUint(uint64(s.id), 10))
 	s.run()
+	close(errChan)
```
[the remark this answers is not in the document]
Comment (fyrchik) [reply pkg/network/server.go:199]: I haven't found where it is shared. In `startServer` from CLI it is used only for network, same for tests.

File: pkg/network/server_test.go (hunk 1 above)
Comment (fyrchik) [reply pkg/network/server_test.go:126]: Fixed.

File: pkg/network/server.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (roman-khimov) [reply pkg/network/server.go:199]: CLI `startServer`:
```
        go serv.Start(errChan)
        go rpcServer.Start(errChan)
```
Same channel.

[the remark this answers is not in the document]
Comment (fyrchik) [reply pkg/network/server.go]: Missed that, fixed.

**Comment by roman-khimov** [comment]
It's too red to be merged.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**